model/src/main/java/ru/training/karaf/repo/BookDescriptionRepoImpl.java

```diff
@@ -0,0 +1,75 @@
+package ru.training.karaf.repo;
+
+import org.apache.aries.jpa.template.JpaTemplate;
+import ru.training.karaf.model.*;
+
+import java.util.Collection;
+
+public class BookDescriptionRepoImpl implements BookDescriptionRepo {
+
+    private JpaTemplate template;
+
+
+    public BookDescriptionRepoImpl(JpaTemplate template) {
+        this.template = template;
+    }
+
+    @Override
+    public void create(long bookId, String description, String date) {
+        BookDescriptionDo descriptionToCreate = new BookDescriptionDo();
+
+        //AuthorDo author = template.txExpr(em -> em.createNamedQuery(AuthorDo.GET, AuthorDo.class).setParameter("name", authorName).getSingleResult());
+
+        //descriptionToCreate.setAuthorDos(author);
+        descriptionToCreate.setDescription(description);
+        descriptionToCreate.setDate(date);
+
+        BookDO bookDO = template.txExpr(em -> em.createNamedQuery(BookDO.GET_BY_ID, BookDO.class).setParameter("id", bookId).getSingleResult());
+        descriptionToCreate.setBook(bookDO);
+
+        bookDO.setDescriptionDo(descriptionToCreate);
+        template.tx(em -> em.merge(bookDO));
+        //template.tx(em -> em.persist(descriptionToCreate));
+
+    }
+
+
+    @Override
+    public void delete() {
+
+    }
+}
+class BookDescriptionImpl implements BookDescription {
+
+    private BookDescriptionDo descriptionDo;
+
+    BookDescriptionImpl(BookDescriptionDo descriptionDo) {
+        if (descriptionDo == null) {
```

Comment:
I think an exception in case passed DO object is null is fine. We should handle such cases on another level. This default object is a bit awkward. It will be hard to trace bugs with it. For example, if you were to find a book with the latest date such default objects give you unexpected results.